### PR TITLE
Add CI check for downstream doc building

### PR DESCRIPTION
**Pull request by mikemhenry**

### CODE REVIEW

**Comment by mikemhenry** [comment]
Sweet, this is working in the sense that we are getting the same error in our CI that we see on RTD

**Review by IAlibay — APPROVED** [review]

**Comment by mikemhenry** [comment]
okay now it really gets the same error!

**Comment by atravitz** [comment]
It appears that in the commit where we transitioned to using versioningit [64d77](https://github.com/OpenFreeEnergy/ofe-sphinx-theme/commit/64d77046676940dea3acff82a1c7c5468e9bf6f9) and on, the `theme/` directory is missing from the build location, when building from the remote branch. 

For example, running 

`pip install git+https://github.com/OpenFreeEnergy/ofe-sphinx-theme@64d77046676940dea3acff82a1c7c5468e9bf6f9`

and then 
`ls /Users/atravitz/micromamba/envs/openfe-docs/lib/python3.10/site-packages/ofe_sphinx_theme/`

shows these files: `__init__.py __pycache__ py.typed    tests`


whereas cloning `ofe-sphinx-theme` and then building with `pip install .` results in the following files within `/Users/atravitz/micromamba/envs/openfe-docs/lib/python3.10/site-packages/ofe_sphinx_theme/`:

`__init__.py __pycache__ _version.py data        py.typed    tests       theme`

I have yet to figure out why `version.py`, `data/`, and `theme/` are omitted with the migration to versioningit.

**Comment by atravitz** [comment]
@mikemhenry fixed! I had to update the MANIFEST.in. Now all the same files are included as in the last-working-commit (`a45f3e`). We probably want to exclude tests, but that can happen in a follow-up PR. 

I'm bothered that I don't know the root-cause of this. My best guess is some change in default settings in setuptools once we switched from using `setuptools_scm` to `versioningit`. Thoughts?

**Comment by mikemhenry** [comment]
oh this is super obvious now, setuptools_scm knows what needs to be included since it uses source control, but when we switched to a different tool, we needed to then use a MANIFEST to tell setup tools what files to package.

**Comment by mikemhenry** [comment]
also i can't approve my own PR, @IAlibay can you look at this?

File: .github/workflows/test-doc-building.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-file: openfe/docs/environment.yaml
+          environment-name: test-env
+
+      - name: Install ofe-sphinx-theme
+        run: |
+          # Remove the version we pulled in from the env and test
+          # the version on this PR
+          pip uninstall ofe-sphinx-theme --yes
```
Comment (IAlibay) [inline .github/workflows/test-doc-building.yaml:53]:
```suggestion
          python -m pip uninstall ofe-sphinx-theme --yes
```

**Review by IAlibay — APPROVED** [review]
Just the one tiny thing, otherwise lgtm!

**Pull request merged**